modules/api/src/main/java/org/apache/ignite/table/mapper/MapperBuilder.java

```diff
@@ -37,93 +50,205 @@
+     *
+     * @param fieldName Field name.
+     * @return Field name for chaining.
+     * @throws IllegalArgumentException If field is {@code null} or class has no declared field with given name.
+     */
+    private String requireValidField(String fieldName) {
+        try {
+            if (fieldName == null || targetType.getDeclaredField(fieldName) == null) {
+                throw new IllegalArgumentException("Mapping for a column already exists: " + fieldName);
+            }
+        } catch (NoSuchFieldException e) {
+            throw new IllegalArgumentException(
+                    String.format("Field not found for class: field=%s, class=%s", fieldName, targetType.getName()));
+        }
+
+        return fieldName;
+    }
+
+    /**
+     * Maps a field to a column.
+     *
+     * @param fieldName        Field name.
+     * @param columnName       Column name.
+     * @param fieldColumnPairs Vararg that accepts (fieldName, columnName) pairs.
+     * @return {@code this} for chaining.
+     * @throws IllegalArgumentException If a field name has not paired column name in {@code fieldColumnPairs}, or a column was already
+     *                                  mapped to another field.
+     * @throws IllegalStateException    if tries to reuse the builder after a mapping has been built.
+     */
+    public MapperBuilder<T> map(@NotNull String fieldName, @NotNull String columnName, String... fieldColumnPairs) {
+        ensureNotStale();
 
-        if (columnToFields.put(Objects.requireNonNull(columnName), Objects.requireNonNull(fieldName)) != null) {
+        if (columnToFields == null) {
+            throw new IllegalArgumentException("Natively supported types doesn't support field mapping.");
+        } else if (fieldColumnPairs.length % 2 != 0) {
+            throw new IllegalArgumentException("Missed a column name, which the field is mapped to.");
+        } else if (columnToFields.put(Objects.requireNonNull(columnName), requireValidField(fieldName)) != null) {
             throw new IllegalArgumentException("Mapping for a column already exists: " + columnName);
         }
 
+        for (int i = 0; i < fieldColumnPairs.length; i += 2) {
+            if (columnToFields.put(Objects.requireNonNull(fieldColumnPairs[i + 1]), requireValidField(fieldColumnPairs[i])) != null) {
+                throw new IllegalArgumentException("Mapping for a column already exists: " + columnName);
+            }
+        }
+
         return this;
     }
 
     /**
-     * Map a field to a type of given class.
+     * Maps a field to a column with using type converter. The value will be converted before write to and after read from column using
+     * provided converter.
      *
-     * @param fieldName   Field name.
-     * @param targetClass Target class.
-     * @return {@code this} for chaining.
+     * @param <ObjectT>  MUST match the object field type, if the individual field mapped to given column.
```

Comment:
Value type. Must match the object field type if the individual field is mapped to a given column.